ksqldb-engine/src/test/java/io/confluent/ksql/function/udf/array/ArrayUnionTest.java

```diff
@@ -0,0 +1,104 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and limitations under the
+ * License.
+ */
+
+package io.confluent.ksql.function.udf.array;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.nullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.contains;
+
+import com.google.common.collect.ImmutableMap;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import org.junit.Test;
+
+public class ArrayUnionTest {
+
+  private final ArrayUnion udf = new ArrayUnion();
+
+  @Test
+  public void shouldUnionArraysOfLikeType() {
+    final List<String> input1 = Arrays.asList("foo", " ", "bar");
+    final List<String> input2 = Arrays.asList("baz");
+    final List<String> result = udf.union(input1, input2);
+    assertThat(result, contains("foo", " ", "bar", "baz"));
+  }
+
+  @Test
+  public void shouldReturnDistinctValues() {
+    final List<String> input1 = Arrays.asList("foo", "foo", "bar");
+    final List<String> input2 = Arrays.asList("baz", "foo");
+    final List<String> result = udf.union(input1, input2);
+    assertThat(result, contains("foo", "bar", "baz"));
+  }
+
+  @SuppressWarnings("unchecked")
+  @Test
+  public void shouldIntersectArraysOfMaps() {
+    final Map<String, Integer> map1 = ImmutableMap.of("foo", 1, "bar", 2, "baz", 3);
+    final Map<String, Integer> map2 = ImmutableMap.of("foo", 10, "baz", 3);
+    final Map<String, Integer> map3 = ImmutableMap.of("foo", 1, "bar", 2, "baz", 3);
+    final List<Map<String, Integer>> input1 = Arrays.asList(map1, map2);
+    final List<Map<String, Integer>> input2 = Arrays.asList(map2, map3);
+    final List<Map<String, Integer>> result = udf.union(input1, input2);
+    assertThat(result, contains(map1, map2));
+  }
+
+  @Test
+  public void shouldUnionArraysContainingNulls() {
+    final List<String> input1 = Arrays.asList(null, "bar");
+    final List<String> input2 = Arrays.asList("foo");
+    final List<String> result = udf.union(input1, input2);
+    assertThat(result, contains(null, "bar", "foo"));
+  }
+
+  @Test
+  public void shouldUnionArraysBothContainingNulls() {
+    final List<String> input1 = Arrays.asList(null, "foo", "bar");
+    final List<String> input2 = Arrays.asList("foo", null);
+    final List<String> result = udf.union(input1, input2);
+    assertThat(result, contains((String) null, "foo", "bar"));
+  }
+
+  @Test
+  public void shouldReturnNullForArraysOfOnlyNulls() {
+    final List<String> input1 = Arrays.asList(null, null);
+    final List<String> input2 = Arrays.asList(null, null, null);
+    final List<String> result = udf.union(input1, input2);
+    assertThat(result, contains(nullValue()));
```

Comment:
This is a bit confusing behavior for me. Why does the test right above contain a null value in the resulting array but this test returns null?

Reply:
there's a subtlety here - in this case the udf doesn't really return null, instead it returns _an array containing a single null_ - which i think is what the function should do in this case, on the grounds that the only distinct value found in either of the input arrays is a null.  This is different than the case that we pass in an actual null, as compared to array-of-nulls which is what we have here, and that other case is tested below.
Does that make sense, or am i misunderstanding the question ?

Reply:
Oh sorry, I just noticed the `contains`. I thought it was `equals` that's why I asked. Your answer and the code totally make sense. Sorry about that

Reply:
yeah, this new matcher-naming throws me off regularly too - i think i liked "contains" much better as something like "isArrayContaining", especially when you're scanning through a whole set of tests  :)